### PR TITLE
Fix application of augeas changes on a Centos7/RHEL7 machine. 

### LINKED ISSUE #11: pam no longer supports @include?
after I ram the module, these started to show up in the logs:

```
Jun 23 10:05:44 mon-app2 sshd[19048]: PAM (sshd) illegal module type: @include
Jun 23 10:05:44 mon-app2 sshd[19048]: PAM pam_parse: expecting return value; [...google-authenticator-all-users]
Jun 23 10:05:44 mon-app2 sshd[19048]: PAM (sshd) no module name supplied
Jun 23 10:05:44 mon-app2 sshd[19048]: pam_unix(sshd:auth): check pass; user unknown
```

could it be that pam on centos/RHEL 7 no longer supports @include lines?!

### PR DESCRIPTION
**Pull request by bekarau**
#### Pull Request (PR) description
Fixes the application of augeas changes on a Centos7/RHEL7 installation.

#### This Pull Request (PR) fixes the following issues
Fixes #11 

#### Further comments
As I've not tested the newer augeas changes against older releases, I've put an explicit test for major release 7 or greater (assuming it won't change in the future either).

### CODE REVIEW

File: manifests/pam/redhat.pp [hunk 1]
```diff
@@ -20,16 +20,33 @@
...
-        context => "/files/etc/pam.d/${name}",
-        changes => [
-          # Purge existing entries
-          'rm include[. =~ regexp("google-authenticator.*")]',
-          "ins include after ${lastauth}",
-          "set include[. = ''] '${rule}'",
-          ],
-        require => File["/etc/pam.d/${rule}"],
-        notify  => Service['sshd'],
+      if ($::facts['os']['release']['major'] >= 7) {
```
Comment (bastelfreak) [inline manifests/pam/redhat.pp]: please remove `::`

**Comment by bastelfreak** [comment]
Thanks for the PR! Please take a look at the failing travis tests.

File: manifests/pam/redhat.pp (hunk 1 above)
Comment (bekarau) [reply manifests/pam/redhat.pp]: Should now be done.

**Comment by bastelfreak** [comment]
there are still some travis tests failing, please have a look. Also please check the email address used in the commits, it is not associated with your github account.

**Comment by bekarau** [comment]
Work vs Home email. Added work email, so should be there now.

**Comment by bastelfreak** [comment]
Thanks!

File: manifests/pam/redhat.pp [hunk 2]
```diff
@@ -20,16 +20,33 @@
...
-        context => "/files/etc/pam.d/${name}",
-        changes => [
-          # Purge existing entries
-          'rm include[. =~ regexp("google-authenticator.*")]',
-          "ins include after ${lastauth}",
-          "set include[. = ''] '${rule}'",
-          ],
-        require => File["/etc/pam.d/${rule}"],
-        notify  => Service['sshd'],
+      if ($facts['os']['release']['major'] >= 7) {
```
Comment (raphink) [inline manifests/pam/redhat.pp]: Maybe this should be restricted to RedHat?

Comment (bastelfreak) [reply manifests/pam/redhat.pp]: good catch. @bekarau what do you think?

Comment (bekarau) [reply manifests/pam/redhat.pp]: The restriction to redhat is done in pam.pp; where this module is called from:
```
  case $::operatingsystem {
    /Debian|Ubuntu/ : {
        googleauthenticator::pam::debian {$name:
          ensure => $ensure,
          mode   => $mode,
        }
      }
    /RedHat|CentOS/ : {
        googleauthenticator::pam::redhat {$name:
          ensure => $ensure,
          mode   => $mode,
        }
      }
    default         : { fail("not supported on ${::operatingsystem}") }
  }
```

Comment (raphink) [reply manifests/pam/redhat.pp]: Ah, my bad, I didn't see this was in `redhat.pp`!

Comment (bekarau) [reply manifests/pam/redhat.pp]: :) no problem

**Pull request merged**